File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo021100Activity.java
package com.liushu.crazyandroid.ui.stage01.chapter02.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.liushu.crazyandroid.R;

import java.text.SimpleDateFormat;
import java.util.Date;

public class Demo021100Activity extends AppCompatActivity {
    private int mGroupPosition;
    private int mChildPosition;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_demo021100);
        mGroupPosition = getIntent().getIntExtra("groupPosition", -1);
        mChildPosition = getIntent().getIntExtra("childPosition", -1);
        initView(mChildPosition);
    }

    private void initView(int childPosition) {
        switch (childPosition) {
            case 0:
            initData0();
                break;
            case 1:

                break;
            case 2:

                break;
            case 3:

                break;
        }
    }

    private void initData0() {

        long deadLine=Long.parseLong("1488210720000");
        SimpleDateFormat dateFormat=new SimpleDateFormat("yyyy.MM.dd HH:mm");
        Date date=new Date(deadLine);
        String endTime = dateFormat.format(date);

        Log.e("22222",deadLine+":"+endTime);

    }
}

File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo021100Activity.kt
package com.liushu.crazyandroid.ui.stage01.chapter02.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.liushu.crazyandroid.R
import java.text.SimpleDateFormat
import java.util.*

class Demo021100Activity : AppCompatActivity() {
    private var mGroupPosition: Int = 0
    private var mChildPosition: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_demo021100)
        mGroupPosition = intent.getIntExtra("groupPosition", -1)
        mChildPosition = intent.getIntExtra("childPosition", -1)
        initView(mChildPosition)
    }

    private fun initView(childPosition: Int) {
        when (childPosition) {
            0 -> initData0()
            1 -> {
            }
            2 -> {
            }
            3 -> {
            }
        }
    }

    private fun initData0() {

        val deadLine = java.lang.Long.parseLong("1488210720000")
        val dateFormat = SimpleDateFormat("yyyy.MM.dd HH:mm")
        val date = Date(deadLine)
        val endTime = dateFormat.format(date)
    }
}
